spec/datasets/data_sources/definitions.yaml

```diff
@@ -0,0 +1,163 @@
+      - array<string>
+      - array<alias>
+      - array<synonym>
+      - sequence
+  DataSource:
+    title: Data Source
+    description: |
+      A description of an external data source.
+      Data sources are external entities, usually governed and managed by an organization,
+      with data repositories, data policies, and specific data structures.
+      (e.g. taxonomy, ontology), individual data sources may provide data beyond this
+      general structure; the data source definition must provide a description
+      of those additional fields so that they can be utilized by interfaces.
+    type: object
+    additionalProperties: false
+    required:
+      [
+        ns,
+        type,
+        title,
+        short_title,
+        data_url,
+        home_url,
+        logo_url,
+        license,
+        item_link,
+        citation,
+        additional_fields,
+      ]
+    properties:
+      ns:
+        type: string
+        format: regex
+        pattern: "^[a-z0-9]+[a-z0-9_]*[a-z0-9]$"
+        description: The RE namespace id for the data source
+        examples:
+          - ncbi_taxonomy
+          - go_ontology
+      type:
+        type: string
+        description: The type of RE namespace and data source
+        examples:
+          - taxonomy
+          - ontology
+      title:
+        type: string
+        description: The fully spelled out title of the organization which manages the data source
+        examples: National Center for Biotechnology Information
+      short_title:
+        type: string
+        description: The shortened name of the organization, e.g. initialization, acronym
+        examples:
+          - NCBI
+      data_url:
+        type: string
+        description: A URL which points to a data repository at which the data loaded into the database is available.
+        examples:
+          - ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/
+      home_url:
+        type: string
+        description: A URL which points to the home page for the organization responsible for the data source
+        examples:
+          - https://www.ncbi.nlm.nih.gov/taxonomy
+      logo_url:
+        type: string
+        description: A URL which points to an image which serves as the logo for the organization responsible for the data source.
+        examples:
+          - https://ci.kbase.us/ui-assets/images/third-party-data-sources/ncbi/logo-51-64.png
+      license:
+        description: An optional url and label which points to the license
+        oneOf:
+          - type: "null"
+          - type: object
+            additionalProperties: false
+            required: [url, label]
+            properties:
+              url:
+                type: string
+                examples:
+                  - http://creativecommons.org/licenses/by-sa/4.0/
+              label:
+                type: string
+                examples:
+                  - Creative Commons Attribution-ShareAlike 4.0 International License
```

Comment:
do these need to be outdented so they are on the same level as `description`? Same with the `item_link` below.

Reply:
No, these are embedded in oneOf - the type is either null or the defined object. Unless I'm misunderstanding the question?

Reply:
No, it was just a question.